R/stroke.R

```diff
@@ -304,6 +316,11 @@ merge_lines <- function(
     while (TRUE) {
       # one segment can appear in multiple strokes when using from_edge
       if (is.na(link) || (is_segment_used[link] && is.null(from_edge))) break
+
+      # store the edge ID and the stroke ID
+      edge_id <- edge_ids[link]
+      stroke_ids[edge_id] <- stroke_id
```

Comment:
I love one-liners :)
```suggestion
      # store the stroke ID
      stroke_ids[edge_ids[link]] <- stroke_id
```